project/Dependencies.scala

```diff
@@ -6,21 +6,32 @@ import scala.collection.Seq
 import sbt._
 
 object Dependencies {
```

Comment:
Given that this is a fresh project, I'm going to be a bit of a stickler: don't put dependencies here. Put them into **build.sbt**. There's literally no reason to put dependencies into a `.scala` file, and there are a *huge* set of ways in which SBT behaves much better when they're in the **build.sbt** instead of in a **.scala** file.

As far as I can tell, this `project/Dependencies.scala` convention comes from Josh Suereth's SBT tutorial back in like, SBT 0.12, and he was only using it as an example (he puts his dependencies into his **build.sbt** and he pretty much always has). If it weren't for the fact that it's a relatively minor yak, I would have already shaved it out of all of our Scala projects. But since this is a new project, I would rather proactively shave and fix things ahead of time.

Reply:
Happy to change, although I did like the deps being in their own file. I wasn't aware of the downsides of putting it in a separate file. Can you give an example?

Reply:
@djspiewak Just want to confirm before I do something which is not the intention: 
if I understand you correctly this means that `AssembleDatasource.scala` also needs to be migrated to `build.sbt`?

Reply:
@rintcius Uh, no that one is probably fine. I haven't looked at it specifically, but it sounds alright. Basically, pushing code into `.scala` files is alright if it's actually *code* and not *configuration*. And if it's *code*, consider making it an `AutoPlugin`, but at least make sure it's code. Configuration is something that needs to be changed and reloaded and introspected much more deeply than script actions do (or can be). Hence why it makes more sense in a `.sbt` file. This is actually *the reason* `.sbt` files exist as a thing in the first place.

Reply:
@djspiewak hmm, how do I need to go about changing `Dependencies.datasourceCore` then (which is currently part of `AssembleDatasource.scala`)?  I mean I don't know how to change 

```
  val setAssemblyKey =
    assembleDatasource  in Compile := { ...
```

so that it accepts a parameter.

Also it doesn't sound right if `AssembleDatasource.scala` refers back to `build.sbt`.

Reply:
If you change `val setAssemblyKey` into a `def` which takes the value of `datasourceCore` as a parameter, it should work.

Reply:
Oh yeah indeed, that's easier than expected